### DISCUSSION TITLE
setInterval can't be cleared

### DISCUSSION THREAD

**Discussion by alifsuria** [GrapesJS/grapesjs#4931] [Q&A] [answered]
Hello @All,

I'm trying to create a countdown component which is then added to the setInterval execution but when the component property is updated I'm trying to clear the previous interval but instead of that the previous interval is still executing stacking with the new interval executing again. I even try to add the interval into the property but still doesn't work.

`editor.DomComponents.addType("countdown-component", {
        model: {
            defaults: {
                timer: null,
                script: function () {
                    let timer;
                    timer = setInterval(function () {
                        console.log('helloworld')
                    }, 1000)
                    console.log(timer)
                },
            },
        },
    })`
Can someone guide me on this? I'm stuck at this for a week.

**Comment by artf** [comment, marked as answer]
I'd suggest to check the [countdown plugin](https://github.com/GrapesJS/components-countdown) code [here](https://github.com/GrapesJS/components-countdown/blob/0bda33b7a7267c8d88e37bbe5964197b98d87234/src/index.ts#L148-L149) which basically overcomes the same issue (the interval is assigned and cleared from the element itself inside the script)
